### DISCUSSION TITLE
MicroPython base USB power

### DISCUSSION THREAD

**Discussion by shaoziyang** [micropython/micropython#9724] [Show and tell]
We have developed a portable programmable power supply powered by USB type-c, which base on microython, and we hope it can bring more help in development and DIY.

![](https://github.com/makediy/PPPS/raw/main/device.jpg)

## Feature

- USB Type-C power supply with PD support
- step up/ step down mode
- Controlled by USB, Bluetooth, Wifi etc.
- Support remote control
- record runtime data and upload to the cloud
- Modular design
- high efficiency with Small size
- Secondary development using **micropython** / circuitpython
- Low cost
- OSHW

Project repository:
https://github.com/makediy/PPPS

**Comment by davefes** [comment]
Printed circuit board as a front panel ... novel idea.  Is this unit in production?

**Reply by shaoziyang** [reply]
This is our second generation prototype, more devices will be built soon. We plan to design a series of tools for engineers and enthusiasts to add IoT functions to traditional tools. It is the first one.

**Comment by Wind-stormger** [comment]
Just ask casually, does it have the function of short circuit protection and information prompt?

**Reply by shaoziyang** [reply]
Yes, it has full protection, supports CC / CV mode.
